activejob/lib/active_job/queue_adapter.rb

```diff
@@ -1,32 +1,41 @@
 require 'active_job/queue_adapters/inline_adapter'
+require 'active_support/core_ext/class/attribute'
 require 'active_support/core_ext/string/inflections'
 
 module ActiveJob
   # The <tt>ActiveJob::QueueAdapter</tt> module is used to load the
   # correct adapter. The default queue adapter is the :inline queue.
   module QueueAdapter #:nodoc:
-    extend ActiveSupport::Concern
+    # We can't use ActiveSupport::Concern here because we need to `prepend`
+    # ClassMethods rather than `extend`ing it.
+    def self.included(klass)
+      klass.class_attribute :queue_adapter, instance_accessor: false, instance_predicate: false
+      klass.singleton_class.prepend(ClassMethods)
```

Comment:
Rather than use an attribute with the same name as public API, pick an internal name, e.g. `:_queue`, and
```ruby
alias_method :queue_adapter, :_queue
def queue_adapter=(name_or_adapter_or_class) self._queue = interpret_adapter(name_or_adapter_or_class) end
```